src/System.Text.Json/src/System/Text/Json/Utf8JsonReader.cs

```diff
@@ -199,6 +201,7 @@ private void StartObject()
 
             _bitStack.PushTrue();
 
+            ValueSpan = _buffer.Slice(_consumed, 1);
```

Comment:
Is this fixing a bug in Utf8JsonReader?

Reply:
Yes.

Previously, I would only update `ValueSpan` (and `ValueSequence`) only when it mattered (i.e. when it was expected for the user to access that property and care about its value - string, number, boolean, etc.). We should probably update ValueSpan and ValueSequence in all cases so that if someone accidentally reads its value they don't get gibberish or the previous token value.

Atm, if you access ValueSpan on a StartObject/EndObject token type, you will get the value from the previous JSON value because we never clear it.